### ISSUE TITLE
[rancher-agent] panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by omegarus** [rancher/rancher#6910]
**Rancher Version:**
Server v1.2.0 Agent v1.1.0
**Docker Version:**
1.11.2
**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu14.04 VMware
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher with two hosts
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Kuberenetes
**Steps to Reproduce:**
Deploy some stack from  catalog and wait.
Sometimes i have problem to resolve service name to of the pod to ip from another pod, but i don't know if its some how linked to this error.
```
time="2016-12-04T08:36:46Z" level=info msg="Received event: Name: compute.instance.activate, Event Id: ce116566-56d8-486e-9815-5f66656b79a7, Resource Id: 1ihm108"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x57a114]

goroutine 104 [running]:
panic(0x834aa0, 0xc42000c070)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/rancher/agent/core/compute.setupNetworkMode(0x5, 0xc42020e340, 0x6, 0x158c8e412a0, 0xc4201b4880, 0x40, 0x0, 0xc420104870, 0x3, 0x3, ...)
        /go/src/github.com/rancher/agent/core/compute/compute_unix.go:189 +0x364
github.com/rancher/agent/core/compute.setupNetworking(0x5, 0xc42020e340, 0x6, 0x158c8e412a0, 0xc4201b4880, 0x40, 0x0, 0xc420104870, 0x3, 0x3, ...)
        /go/src/github.com/rancher/agent/core/compute/compute_unix.go:101 +0xd2
github.com/rancher/agent/core/compute.DoInstanceActivate(0x5, 0xc42020e340, 0x6, 0x158c8e412a0, 0xc4201b4880, 0x40, 0x0, 0xc420104870, 0x3, 0x3, ...)
        /go/src/github.com/rancher/agent/core/compute/compute.go:81 +0xae1
github.com/rancher/agent/handlers.(*ComputeHandler).InstanceActivate(0xc420338900, 0xc42010a000, 0xc42033a000, 0x3, 0x3)
        /go/src/github.com/rancher/agent/handlers/compute.go:44 +0x34a
github.com/rancher/agent/handlers.(*ComputeHandler).InstanceActivate-fm(0xc42010a000, 0xc42033a000, 0xc4201cfc98, 0x3)
        /go/src/github.com/rancher/agent/handlers/common.go:32 +0x3e
github.com/rancher/agent/handlers.logRequest.func1(0xc42010a000, 0xc42033a000, 0xc4202ced60, 0x5e1101)
        /go/src/github.com/rancher/agent/handlers/common.go:49 +0x14a
github.com/rancher/agent/handlers.cleanLog.func1(0xc42010a000, 0xc42033a000, 0xc42018a200, 0x19)
        /go/src/github.com/rancher/agent/handlers/common.go:55 +0x59
github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events.doWork(0xc42010a000, 0xc420387ad0, 0xc42033a000, 0xcb9ac0, 0xc420168580)
        /go/src/github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events/worker.go:94 +0x2a4
github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events.(*skippingWorkerPool).HandleWork.func1(0xc420390640, 0x1, 0xc42010a000, 0xc420387ad0, 0xc42033a000)
        /go/src/github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events/worker.go:47 +0x9e
created by github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events.(*skippingWorkerPool).HandleWork
        /go/src/github.com/rancher/agent/vendor/github.com/rancher/event-subscriber/events/worker.go:48 +0x19f
INFO: Starting agent for 41681DDF25E52D8D08A8

```

**Comment by aemneina** [comment]
That stacktrace looks like there's an issue communicating with one of your hosts. Are they all in 'active' state in the hosts screen, when this happens?

**Comment by omegarus** [comment]
@aemneina Yes always in active state.
I double checked all network setting and test with iperf between all 3 servers. No packet drops. Everything looks good.

**Comment by aemneina** [comment]
@StrongMonkey any idea on what could be going on here?

**Comment by StrongMonkey** [comment]
this is because we are launching a container with network mode `container` but there is no data in its network container(stopped or removed).

**Comment by StrongMonkey** [comment]
wondering it is a bug in cattle. The data are sent by cattle so if network mode is `container`, the data should contain its network container.

**Comment by justsml** [comment]
@StrongMonkey is this a v1.2 issue? Or affected by the new ver?

**Comment by StrongMonkey** [comment]
@justsml fixed by this PR https://github.com/rancher/agent/pull/107. And i believe the issue you referenced is not related to this one.

**Comment by justsml** [comment]
Thanks @StrongMonkey ❤️

**Comment by galal-hussein** [comment]
I tested multiple containers with network-mode set to `container` and i can't see the panic, @omegarus can you please retest your workflow and see if the problem still happening or not

**Comment by omegarus** [comment]
@galal-hussein I am going to retest it today with Rancher server v1.2.1-rc2 and agent-v1.1.1-rc1.

**Comment by omegarus** [comment]
@galal-hussein After few hours of testing looks like problem is fixed.
However I still have problem with dns resolutions of the services.  Issue 7012.

**Comment by deniseschannon** [comment]
Both @omegarus and @galal-hussein reported that this has been fixed in v1.2.1. Closing the issue.
